src/lib.rs

```diff
@@ -53,15 +43,12 @@ pub fn candidate_cert_dirs() -> impl Iterator<Item = &'static Path> {
     .filter(|p| p.exists())
 }
 
-/// Deprecated as this isn't sound, use [`init_openssl_env_vars`] instead.
-#[doc(hidden)]
-#[deprecated(note = "this function is not safe, use `init_openssl_env_vars` instead")]
-pub fn init_ssl_cert_env_vars() {
-    unsafe {
-        init_openssl_env_vars();
-    }
-}
-
+#[cfg(feature = "allow-unsafe-set-env")]
```

Comment:
A nitpick: I think `#[cfg]` gates and other attributes usually come after the documentation, so they're not separated from the code by a screenfull of text.